src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> addedNode = new Node<>(value);
+        if (size == 0) {
+            first = addedNode;
+        } else {
+            addedNode.prev = last;
+            last.next = addedNode;
+        }
+        last = addedNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAdding(index);
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> addedNode = new Node<>(value);
+            addedNode.next = first;
+            first.prev = addedNode;
+            first = addedNode;
+            size++;
+        } else {
+            Node<T> searchedNode = findNodeByIndex(index);
+            Node<T> addedNode = new Node<>(value);
+            addedNode.next = searchedNode;
+            addedNode.prev = searchedNode.prev;
+            searchedNode.prev.next = addedNode;
+            searchedNode.prev = addedNode;
+            size++;
+        }
```

Comment:
let's return early here -on line 28, and unwrap the rest of the code to make it more readable. We can also extract some common logic from two clauses:
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkIndex(index);
        Node<T> addedNode = new Node<>(value);
        if (index == 0) {
            addedNode.next = first;
            first.prev = addedNode;
            first = addedNode;
        } else {
            Node<T> searchedNode = findNodeByIndex(index);
            addedNode.next = searchedNode;
            addedNode.prev = searchedNode.prev;
            searchedNode.prev.next = addedNode;
            searchedNode.prev = addedNode;
        }
        size++;
```
Here, we can also do a little trick and place index checker after index == size, and this way we will eliminate the need for special checker for add method :)